dtfs-central-api/src/v1/controllers/tfm/amendments/tfm-get-amendments.controller.js

```diff
@@ -324,6 +324,31 @@ const findLatestCompletedAmendmentByFacilityId = async (facilityId) => {
 };
 exports.findLatestCompletedAmendmentByFacilityId = findLatestCompletedAmendmentByFacilityId;
 
+// finds the latest completed amendment and returns the version
+const findLatestCompletedAmendmentByFacilityIdVersion = async (facilityId) => {
+  if (ObjectId.isValid(facilityId)) {
+    const { COMPLETED } = CONSTANTS.AMENDMENT.AMENDMENT_STATUS;
+
+    try {
+      const collection = await db.getCollection('tfm-facilities');
+      const amendment = await collection.aggregate([
+        { $match: { _id: ObjectId(facilityId) } },
+        { $unwind: '$amendments' },
+        { $match: { 'amendments.status': COMPLETED } },
+        { $sort: { 'amendments.updatedAt': -1, 'amendments.version': -1 } },
+        { $project: { _id: 0, amendments: 1 } },
+        { $limit: 1 }
+      ]).toArray();
+      return amendment[0]?.amendments?.version ?? null;
```

Comment:
Can we check whether `amendment` has an data inside an array using `amendment.length > 0` please?

Reply:
We shouldn't need that though. If `amendment[0]` fails, then it will return the value after `??` - in this case `null`